### PR TITLE
Add back Microsoft.CodeQuality.Analyzers.md

### LINKED ISSUE #1285: The Microsoft.CodeQuality.Analyzers "more info" link is 404
The More info link on the Readme for `Microsoft.CodeQuality.Analyzers` leads to a 404 page on Github.

#### Analyzer package
N/A

#### Analyzer
N/A

#### Repro steps
N/A

#### Expected behavior
N/A

#### Actual behavior
N/A

### PR DESCRIPTION
**Pull request by mavasani**
Also update the hyperlink in the repo's README.md.
Fixes #1285

### CODE REVIEW

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+
+Category: Design
+
+Severity: Warning
+
+Help: [http://msdn.microsoft.com/library/ms182172.aspx](http://msdn.microsoft.com/library/ms182172.aspx)
+
+### CA1003: Use generic event handler instances ###
+
+A type contains a delegate that returns void, whose signature contains two parameters (the first an object and the second a type that is assignable to EventArgs), and the containing assembly targets Microsoft .NET Framework?2.0.
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: > ?

That's probably supposed to be a space.

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+To broaden the usability of a collection, implement one of the generic collection interfaces. Then the collection can be used to populate generic collection types.
+
+Category: Design
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/ms182132.aspx](https://msdn.microsoft.com/en-us/library/ms182132.aspx)
+
+### CA1012: Abstract types should not have constructors ###
+
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: No description?

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+ComVisibleAttribute determines how COM clients access managed code. Good design dictates that assemblies explicitly indicate COM visibility. COM visibility can be set for the whole assembly and then overridden for individual types and type members. If this attribute is not present, the contents of the assembly are visible to COM clients.
+
+Category: Design
+
+Severity: Warning
+
+Help: [http://msdn.microsoft.com/library/ms182157.aspx](http://msdn.microsoft.com/library/ms182157.aspx)
+
+### CA1018: Mark attributes with AttributeUsageAttribute ###
+
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: No description

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+Category: Design
+
+Severity: Warning
+
+Help: [http://msdn.microsoft.com/library/ms182158.aspx](http://msdn.microsoft.com/library/ms182158.aspx)
+
+### CA1019: Define accessors for attribute arguments ###
+
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: No description

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md [hunk 5]
```diff
@@ -0,0 +1,547 @@
...
+Types are declared in namespaces to prevent name collisions and as a way to organize related types in an object hierarchy.
+
+Category: Design
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/ms182134.aspx](https://msdn.microsoft.com/en-us/library/ms182134.aspx)
+
+### CA1052: Static holder types should be Static or NotInheritable ###
+
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: No description

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md [hunk 6]
```diff
@@ -0,0 +1,547 @@
...
+A method that is not expected to throw exceptions throws an exception.
+
+Category: Design
+
+Severity: Warning
+
+Help: [https://msdn.microsoft.com/en-us/library/bb386039.aspx](https://msdn.microsoft.com/en-us/library/bb386039.aspx)
+
+### CA1066: Type {0} should implement IEquatable<T> because it overrides Equals ###
+
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: No description or link? Do we not have a page for this one or the next two?

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md [hunk 7]
```diff
@@ -0,0 +1,547 @@
...
+Help: [https://msdn.microsoft.com/en-us/library/0fss9skc.aspx](https://msdn.microsoft.com/en-us/library/0fss9skc.aspx)
+
+### CA2007: Do not directly await a Task ###
+
+While authoring a library where the consumer is not known and when there is no need for a SynchronizationContext,  consider using ConfigureAwait(false). Otherwise, the consumers of the library may face deadlocks by consuming the asynchronous methods in a blocking fashion.
+
+Category: Reliability
+
+Severity: Warning
+
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: No link?

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md (hunk 6 above)
Comment (mavasani) [reply src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: Seems like next set of rules are not implemented by legacy FxCop, hence no msdn link. I will update these once I tackle https://github.com/dotnet/roslyn-analyzers/issues/1269

Comment (mavasani) [reply src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: I will add the description for these rules in this PR

File: src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md (hunk 7 above)
Comment (mavasani) [reply src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md]: I will update this once I tackle #1269

**Pull request merged**